File: your_project/sample.js
class UserManager {
    /**
     * Manages user accounts and related operations.
     */
    constructor() {
      this.users = [];
    }
  
    addUser(name, age, role = "user") {
      /**
       * Adds a new user to the system.
       */
      if (!name || typeof name !== "string") {
        throw new Error("Name must be a non-empty string.");
      }
      if (age < 0) {
        throw new Error("Age must be a positive number.");
      }
      this.users.push({ name, age, role });
      console.log(`User ${name} added.`); 
    }
  
    removeUser(name) {
      /**
       * Removes a user by their name.
       */
      const userIndex = this.users.findIndex((user) => user.name === name);
      if (userIndex !== -1) {
        this.users.splice(userIndex, 1);
        console.log(`User ${name} removed.`);
      } else {
        // @TODO: Implement error handling for missing users.
        console.log(`User ${name} not found.`); 
      }
    }
  
    sendWelcomeEmail(email) {
      /**
       * Sends a welcome email to the given address.
       */
      if (!email.includes("@")) {
        // @REFACTOR: Use a more robust email validation.
        throw new Error("Invalid email address."); 
      }
      console.log(`Welcome email sent to ${email}!`); // Simulate sending email
    }
  
    listUsers() {
      /**
       * Lists all users.
       */
      console.log("Users in the system:");
      this.users.forEach((user) => {
        console.log(`- ${user.name} (Age: ${user.age}, Role: ${user.role})`);
      });
    }
  }

File: your_project/sample.py

class UserManager:
    """
    Manages user accounts and related operations.
    """

    def __init__(self):
        self.users = [] 

    def add_user(self, name, age, role="user"):
        """
        Adds a new user to the system.
        """
        if not name or not isinstance(name, str):
            raise ValueError("Name must be a non-empty string.")
        if age < 0:
            raise ValueError("Age must be a positive number.")
        self.users.append({"name": name, "age": age, "role": role})
        print(f"User {name} added.") 

    def remove_user(self, name):
        """
        Removes a user by their name.
        """
        # @TODO: Implement error handling for user not found.
        # @REFACTOR: Use a more efficient data structure for lookup.
        user_to_remove = None
        for user in self.users:  
            if user["name"] == name:
                user_to_remove = user
                break
        if user_to_remove:
            self.users.remove(user_to_remove)
            print(f"User {name} removed.")
        else:
            print(f"User {name} not found.")  

    def send_welcome_email(self, email):
        """
        Sends a welcome email to the given address.
        """
        # @REFACTOR: Replace email validation with a proper regex or library.
        if "@" not in email:  
            raise ValueError("Invalid email address.")
        print(f"Welcome email sent to {email}!")  # Simulate sending email

    def list_users(self):
        """
        Lists all users.
        """
        print("Users in the system:")
        for user in self.users:
            print(f"- {user['name']} (Age: {user['age']}, Role: {user['role']})")
